### ISSUE TITLE
Minor improvements in the module

### ISSUE THREAD

**Issue by adekusar-drl** [Qiskit/qiskit-machine-learning#24]
### What is the expected enhancement?
There are minor improvements to do:

- algorithms/init

- [x] NNC/NNR/VQR are missing in docstring

- datasets 

- [ ] Return np.ndarrays, optionally one-hot encoded (remove all functions not used)

- neural_networks/init

- [x] OpflowQNN missing in docstring

- loss_functions 

- [x] Should we rename the file loss.py to loss_functions.py?

- [x] Should we include the actual loss functions in the init.py to make them importable directly, e.g. "from qml.utils.lossfunctions import CrossEntropyLoss"? (currently there is only the copyright header in the loss_functions init)

- [x] Are we using L2LossProbability or should we remove it? Or is it tested somewhere? 

- [ ] We should extend the docstrings of the loss functions a little bit to include what exactly they are computing. 

- [x] Did we test the KLDiv somewhere (it doesn't have a gradient implemented)? 

- [x] Do we need softmax/stable_softmax, doesn't seem to be used anywhere, maybe remove?

**Comment by adekusar-drl** [comment]
Datasets are fixed in #26
